src/client.ts

```diff
@@ -364,14 +364,14 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
   }
 
   public async fetchQualifiedSegments(options?: optimizely.OptimizelySegmentOption[]): Promise<boolean> {
-    if (this.odpExplicitlyOff) {
-      return true;
-    }
-
     if (!this.userContext) {
       return false;
     }
 
+    if (this.odpExplicitlyOff || !this._client?.getProjectConfig()?.odpIntegrationConfig?.integrated) {
```

Comment:
Looks good. "getProjectConfig" is not supposed to be public API. Any way to use OptimizelyConfig, which is public access for the project config.

Reply:
OptimizelyConfig does not contain odp integration information. However it contains the datafile string. Do you recommend parsing that and extract integration info from there? 

Its also worth mentioning that JS SDK has a private `isOdpIntegrated` method of `Optimizely` base class. You can give it a look here - [isOdpIntegrated](https://github.com/optimizely/javascript-sdk/blob/2f6dc05c6aa7aacc8f6e2e99c1da128fc585742a/lib/optimizely/index.ts#L1726)

Do you recommend making it public and consume that from react ?

Reply:
I see it will be better to open optimizely.isOdpIntegrated rather than from getProjectConfig.

Reply:
Ok. This has to wait for the JS change then.